### PR TITLE
Added support for reading body for STS API

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Added support for reading body in STS API

## Motivation and Context
STS API supports both URL query params and reading
from a body.

## Regression
No

## How Has This Been Tested?
Using the following script
```
#!/bin/bash

export AWS_PROFILE=gw
export MINIO_ACCESS_KEY=minio
export MINIO_SECRET_KEY=minio123
export MINIO_IAM_JWKS_URL=http://localhost:9763/oauth2/jwks
export MINIO_ETCD_ENDPOINTS=http://localhost:2379
minio gateway s3
```

And then get the token from 
```
~ aws --endpoint-url=http://localhost:9000 sts assume-role-with-web-identity --role-arn aws:arn:wwffffwfer123123123123 --role-session-name RoleName --web-identity-token eyJ4NXQiOiJOVEF4Wm1NeE5ETXlaRGczTVRVMVpHTTBNekV6T0RKaFpXSTRORE5sWkRVMU9HRmtOakZpTVEiLCJraWQiOiJOVEF4Wm1NeE5ETXlaRGczTVRVMVpHTTBNekV6T0RKaFpXSTRORE5sWkRVMU9HRmtOakZpTVEiLCJhbGciOiJSUzI1NiJ9.eyJzdWIiOiJhZG1pbkBjYXJib24uc3VwZXIiLCJhdWQiOiJOWkxPT0ZSU2x1dzlSZklrdUhHcWZrMUhGcDRhIiwibmJmIjoxNTQ5MDY3MTk4LCJhenAiOiJOWkxPT0ZSU2x1dzlSZklrdUhHcWZrMUhGcDRhIiwiaXNzIjoiaHR0cHM6XC9cL2xvY2FsaG9zdDo5NDQzXC9vYXV0aDJcL3Rva2VuIiwiZXhwIjoxNTQ5MDcwNzk4LCJpYXQiOjE1NDkwNjcxOTgsImp0aSI6ImJkNjZjMGIwLTc4NjAtNDQzZS05MGQwLWQwOTE4NGU3M2VlMCJ9.k9qDm5TuWLyiUiPgbVzkkfrZ2zdGD3LYJ8z2TU1PHd0wphKtk4KobPpLIMmr9PbEZ-2EwSdoMKwwKGH5IWkgygnZniwbuZD7j012O-sNkZWK_YYFrd0R2D5qZDmwGprPa-ZAWkkzV_TFxZjI5VHAkRo49-zXo7wRjNNKyKbFmv0aChhwnA15DcMVXOy94f2RLOZp9t23SvpG6zwIkHwulA_oPrP79j90ruPJt36xwyPjtnjFbQ5Cukc0sg4rdzuoRaA1G9749Jqdt_wJR6udZ8vYHgURU9NEJ76ODXZnvGjvtBcuEjuta-DJVyx7AjiXrb51hdgGkGl5bvVRa4wxqQ --provider-id 123123123
{
    "Credentials": {
        "AccessKeyId": "LCJRZRJ0ULDZ5ULMAO55",
        "SecretAccessKey": "oqHh9RNpfL1l4GKi8ztDOL1andhq2JNfcTO++Xoh",
        "SessionToken": "eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJhY2Nlc3NLZXkiOiJMQ0pSWlJKMFVMRFo1VUxNQU81NSIsImF1ZCI6Ik5aTE9PRlJTbHV3OVJmSWt1SEdxZmsxSEZwNGEiLCJhenAiOiJOWkxPT0ZSU2x1dzlSZklrdUhHcWZrMUhGcDRhIiwiZXhwIjoxNTQ5MDcwNzk4LCJpYXQiOjE1NDkwNjcxOTgsImlzcyI6Imh0dHBzOi8vbG9jYWxob3N0Ojk0NDMvb2F1dGgyL3Rva2VuIiwianRpIjoiYmQ2NmMwYjAtNzg2MC00NDNlLTkwZDAtZDA5MTg0ZTczZWUwIiwibmJmIjoxNTQ5MDY3MTk4LCJzdWIiOiJhZG1pbkBjYXJib24uc3VwZXIifQ.jrAh4gx_UnxyL_HKVVKxGCkqpFkfpCb9HcaXKxeyFKUlOgUOrfuAz3qHTnVyd1IpK-q3uJ83fSubjkjI_3rS-g",
        "Expiration": "2019-02-01T17:26:38-08:00"
    },
    "SubjectFromWebIdentityToken": "admin@carbon.super",
    "AssumedRoleUser": {
        "Arn": ""
    }
}
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: cmd/sts-handlers.go [hunk 1]
```diff
@@ -41,139 +46,86 @@ func registerSTSRouter(router *mux.Router) {
...
+		writeSTSErrorResponse(w, ErrSTSInvalidParameterValue)
 		return
 	}
 
-	// Notify all other Minio peers to reload temp users
-	for _, nerr := range globalNotificationSys.LoadUsers() {
-		if nerr.Err != nil {
-			logger.GetReqInfo(ctx).SetTags("peerAddress", nerr.Host.String())
-			logger.LogIf(ctx, nerr.Err)
-		}
```
Comment (poornas) [inline cmd/sts-handlers.go:130]: @harshavardhana ,it is not clear why the credentials generation and setting temp user credentials in globalIAMSys is being removed in the context of this PR.

**Review by poornas — CHANGES_REQUESTED** [review]

File: cmd/sts-handlers.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/sts-handlers.go:130]: It is moved into a single function @poornas

Comment (harshavardhana) [reply cmd/sts-handlers.go:130]: The code is still present where its supposed to be and how it should be used. you need to check the entire file to see what this PR does.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :x: [more...](#82a9f7f5-fe08-479a-99c2-7e3d226878b6) |
<a name="82a9f7f5-fe08-479a-99c2-7e3d226878b6"></a>
### 7188-b9b0559/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      minikube:30418
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9ff20a61d8c9:/mint/log /tmp/mint-logs'

(1/13) Running awscli tests ... done in 3 minutes and 40 seconds
(2/13) Running aws-sdk-go tests ... done in 0 seconds
(3/13) Running aws-sdk-java tests ... done in 8 seconds
(4/13) Running aws-sdk-php tests ... done in 42 seconds
(5/13) Running aws-sdk-ruby tests ... done in 2 seconds
(6/13) Running mc tests ... done in 21 seconds
(7/13) Running minio-dotnet tests ... FAILED in 2 minutes and 49 seconds

Unhandled Exception: System.AggregateException: One or more errors occurred. (Assert.Fail failed. ) ---> Microsoft.VisualStudio.TestTools.UnitTesting.AssertFailedException: Assert.Fail failed. 
   at Microsoft.VisualStudio.TestTools.UnitTesting.Assert.HandleFail(String assertionName, String message, Object[] parameters)
   at Minio.Functional.Tests.FunctionalTest.<>c.<ListIncompleteUpload_Test2>b__92_1(Exception ex) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 2534
   at System.Reactive.AutoDetachObserver`1.OnErrorCore(Exception exception) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Internal\AutoDetachObserver.cs:line 76
   at System.Reactive.AnonymousSafeObserver`1.OnError(Exception error) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\AnonymousSafeObserver.cs:line 62
   at System.Reactive.Linq.ObservableImpl.Throw`1._.Invoke() in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Linq\Observable\Throw.cs:line 42
   at System.Reactive.Concurrency.Scheduler.Invoke(IScheduler scheduler, Action action) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Concurrency\Scheduler.Simple.cs:line 83
   at System.Reactive.Producer`2.SubscribeRaw(IObserver`1 observer, Boolean enableSafeguard) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Internal\Producer.cs:line 127
   at System.Reactive.Linq.QueryLanguage.<>c__DisplayClass330_0`1.<Create>b__0(IObserver`1 observer) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Linq\QueryLanguage.Creation.cs:line 46
   at System.Reactive.AnonymousObservable`1.SubscribeCore(IObserver`1 observer) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\AnonymousObservable.cs:line 37
   at System.Reactive.ObservableBase`1.ScheduledSubscribe(IScheduler _, AutoDetachObserver`1 autoDetachObserver) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\ObservableBase.cs:line 82
   at System.Reactive.Concurrency.CurrentThreadScheduler.Trampoline.Run(SchedulerQueue`1 queue) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Concurrency\CurrentThreadScheduler.cs:line 126
   at System.Reactive.Concurrency.CurrentThreadScheduler.Schedule[TState](TState state, TimeSpan dueTime, Func`3 action) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\Concurrency\CurrentThreadScheduler.cs:line 94
   at System.Reactive.ObservableBase`1.Subscribe(IObserver`1 observer) in D:\a\1\s\Rx.NET\Source\src\System.Reactive\ObservableBase.cs:line 51
   at Minio.Functional.Tests.FunctionalTest.ListIncompleteUpload_Test2(MinioClient minio) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 2532
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Minio.Functional.Tests.FunctionalTest.Main(String[] args) in /mint/run/core/minio-dotnet/FunctionalTest.cs:line 244

Executed 6 out of 13 tests successfully.
```

File: cmd/sts-handlers.go (hunk 1 above)
Comment (poornas) [reply cmd/sts-handlers.go:130]: my bad - diff was a little misleading.

Comment (harshavardhana) [reply cmd/sts-handlers.go:130]: Yeah it's perhaps one of those times where git fails. The movement of code as file lines is captured but the general essence is lost. This one is a rare example. 

One cannot tell from the diff what it is at first due to how the code moved. Quite interesting..

**Review by poornas — DISMISSED** [review]
LGTM, tested

**Comment by poornas** [comment]
travis failure needs to be checked

**Comment by harshavardhana** [comment]
I don't see a Travis failure, was it for a previous run ?

**Comment by kannappanr** [comment]
I restarted it.

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :x: [more...](#74d9c050-b6cc-4131-9ecd-a21c0c255990) |
| mint-xl.sh | :x: [more...](#aa37b16e-bc6a-4968-92f2-aa3cf9f55694) |
| mint-large-bucket.sh | :x: [more...](#f86c4727-1e93-419c-86ca-ce1088845333) |
| mint-compression-dist-xl.sh | :x: [more...](#d683ad43-7203-4063-8eaf-07a5f1c1b5ec) |
| mint-compression-fs.sh | :x: [more...](#a1865913-d263-43f5-a5b1-92de45773c64) |
| mint-worm.sh | :x: [more...](#cb011989-94d3-420f-b36c-eddf6f4f5c89) |
| mint-fs.sh | :x: [more...](#69d7e20e-27a5-4515-967b-81c1c57b9642) |
| mint-dist-xl.sh | :x: [more...](#b82a6e20-fa4c-4f40-8ec9-b10fe10210c8) |
| mint-gateway-nas.sh | :x: [more...](#935cbc10-e513-46a5-bc94-1bdfec4e4c6a) |
<a name="935cbc10-e513-46a5-bc94-1bdfec4e4c6a"></a>
### 7188-b7b20ef/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      32697:32697
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 20d0df4ccfa6:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-fq9hkeb541xrmt",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 0,
  "error": "RequestError: send request failed\ncaused by: Put http://32697:32697/aws-sdk-go-test-fq9hkeb541xrmt: dial tcp: lookup 32697: no such host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="d683ad43-7203-4063-8eaf-07a5f1c1b5ec"></a>
### 7188-b7b20ef/mint-compression-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      32334:32334
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d8ca610bc67d:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-v3ft9vdaz1o0pg",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 0,
  "error": "RequestError: send request failed\ncaused by: Put http://32334:32334/aws-sdk-go-test-v3ft9vdaz1o0pg: dial tcp: lookup 32334: no such host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="a1865913-d263-43f5-a5b1-92de45773c64"></a>
### 7188-b7b20ef/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      31241:31241
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 09267b9c1669:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-v5lm5c6argpm6o",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 0,
  "error": "RequestError: send request failed\ncaused by: Put http://31241:31241/aws-sdk-go-test-v5lm5c6argpm6o: dial tcp: lookup 31241: no such host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="f86c4727-1e93-419c-86ca-ce1088845333"></a>
### 7188-b7b20ef/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      30765:30765
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 33822d56a3ae:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-qbcqrwstzcnt36",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 0,
  "error": "RequestError: send request failed\ncaused by: Put http://30765:30765/aws-sdk-go-test-qbcqrwstzcnt36: dial tcp: lookup 30765: no such host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="69d7e20e-27a5-4515-967b-81c1c57b9642"></a>
### 7188-b7b20ef/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      31871:31871
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 63e7decf6220:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-j1jhuxxq14n3v3",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 0,
  "error": "RequestError: send request failed\ncaused by: Put http://31871:31871/aws-sdk-go-test-j1jhuxxq14n3v3: dial tcp: lookup 31871: no such host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="74d9c050-b6cc-4131-9ecd-a21c0c255990"></a>
### 7188-b7b20ef/mint-compression-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      31603:31603
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 528a9406b110:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-sgiqyw4z0eebza",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 0,
  "error": "RequestError: send request failed\ncaused by: Put http://31603:31603/aws-sdk-go-test-sgiqyw4z0eebza: dial tcp: lookup 31603: no such host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="aa37b16e-bc6a-4968-92f2-aa3cf9f55694"></a>
### 7188-b7b20ef/mint-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      31240:31240
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 027419721cfe:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-evtqjagpdth015",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 0,
  "error": "RequestError: send request failed\ncaused by: Put http://31240:31240/aws-sdk-go-test-evtqjagpdth015: dial tcp: lookup 31240: no such host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```
<a name="cb011989-94d3-420f-b36c-eddf6f4f5c89"></a>
### 7188-b7b20ef/mint-worm.sh.log:
```
Running with
SERVER_ENDPOINT:      30729:30729
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            worm
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp caf42e633f2d:/mint/log /tmp/mint-logs'

(1/1) Running worm tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "bucket-ae5dazazy72k5j2ls5v6silh4vydwq70adzy7a9bes8qcbcnn9eqdw6",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 0,
  "error": "RequestError: send request failed\ncaused by: Put http://30729:30729/bucket-ae5dazazy72k5j2ls5v6silh4vydwq70adzy7a9bes8qcbcnn9eqdw6: dial tcp: lookup 30729: no such host",
  "function": "PutAndDelete",
  "message": "WORM_MODE ON  - CreateBucket Failed",
  "name": "test worm mode",
  "status": "FAIL"
}

Executed 0 out of 1 tests successfully.
```
<a name="b82a6e20-fa4c-4f40-8ec9-b10fe10210c8"></a>
### 7188-b7b20ef/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      31014:31014
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e7b7e59ff303:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-pynzv3eybkqquc",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 0,
  "error": "RequestError: send request failed\ncaused by: Put http://31014:31014/aws-sdk-go-test-pynzv3eybkqquc: dial tcp: lookup 31014: no such host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 13 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :x: [more...](#112eee93-9b5a-4941-9202-bcc41c122be4) |
<a name="112eee93-9b5a-4941-9202-bcc41c122be4"></a>
### 7188-632107d/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:30533
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 1ead47350944:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... done in 1 seconds
(2/13) Running aws-sdk-java tests ... done in 0 seconds
(3/13) Running aws-sdk-php tests ... done in 41 seconds
(4/13) Running aws-sdk-ruby tests ... FAILED in 1 seconds
{
  "name": "aws-sdk-ruby",
  "function": "presignedPost(bucket_name,file_name,expires_in_sec,max_byte_size)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-3f98284c5827",
    "file_name": "datafile-1-MB",
    "expires_in_sec": 60,
    "max_byte_size": 3145728
  },
  "duration": 55.92,
  "error": "Expected to be created object does NOT exist",
  "status": "FAIL"
}

Executed 3 out of 13 tests successfully.
```

**Comment by harshavardhana** [comment]
Can this be reviewed again ?

**Review by poornas — APPROVED** [review]

**Review by kannappanr — APPROVED** [review]
LGTM. Tested with the steps in the PR.

**Pull request merged**